rust/ql/lib/codeql/rust/frameworks/stdlib/fs.model.yml

```
@@ -16,4 +16,5 @@ extensions:
       - ["lang:std", "<crate::path::PathBuf as crate::convert::From>::from", "Argument[0]", "ReturnValue", "taint", "manual"]
       - ["lang:std", "<crate::path::Path>::join", "Argument[self]", "ReturnValue", "taint", "manual"]
       - ["lang:std", "<crate::path::Path>::join", "Argument[0]", "ReturnValue", "taint", "manual"]
+      - ["lang:std", "<crate::path::Path>::canonicalize", "Argument[self]", "ReturnValue.Field[crate::result::Result::Ok(0)].OptionalStep[normalize-path]", "taint", "manual"]
```

```suggestion
      - ["lang:std", "<crate::path::Path>::canonicalize", "Argument[self].OptionalStep[normalize-path]", "ReturnValue.Field[crate::result::Result::Ok(0)]", "taint", "manual"]
```

Sorry, I put the optional step in the wrong place when I posted the code. It should be at the end of the input, not at the end of the output.

---

When I put the step in the "input" then I cannot get the `isBarrier` predicate to work properly.

---

Yes that works, see also: https://github.com/github/codeql/pull/18960/commits/2804c130279cb74e4c5eee7851535ed998f15e00